R/exclude-pregnancy.R

```diff
@@ -77,9 +81,33 @@ exclude_pregnancy <- function(excluded_pcos, pregnancy_dates, included_hba1c) {
     )
 
   excluded_pcos |>
-    dplyr::full_join(pregnancy_dates, by = dplyr::join_by("pnr")) |>
-    dplyr::full_join(included_hba1c, by = dplyr::join_by("pnr", "date")) |>
-    dplyr::filter(!!criteria) |>
+    dplyr::full_join(
+      included_hba1c,
+      by = dplyr::join_by("pnr", "date")
+    ) |>
+    dplyr::full_join(
```

Comment:
To only join the pregnancy events of people who have any HbA1c or GLD events that need censoring:
```suggestion
    dplyr::left_join(
```